server/src/test/java/io/druid/client/HttpServerInventoryViewTest.java

```diff
@@ -0,0 +1,319 @@
+                        new SegmentChangeRequestsSnapshot(
+                            false,
+                            null,
+                            SegmentChangeRequestHistory.Counter.ZERO,
+                            ImmutableList.of(
+                                new SegmentChangeRequestLoad(segment1)
+                            )
+                        )
+                    )
+                )
+            ),
+            Futures.immediateFuture(
+                new ByteArrayInputStream(
+                    jsonMapper.writeValueAsBytes(
+                        new SegmentChangeRequestsSnapshot(
+                            false,
+                            null,
+                            SegmentChangeRequestHistory.Counter.ZERO,
+                            ImmutableList.of(
+                                new SegmentChangeRequestDrop(segment1),
+                                new SegmentChangeRequestLoad(segment2),
+                                new SegmentChangeRequestLoad(segment3)
+                            )
+                        )
+                    )
+                )
+            ),
+            Futures.immediateFuture(
+                new ByteArrayInputStream(
+                    jsonMapper.writeValueAsBytes(
+                        new SegmentChangeRequestsSnapshot(
+                            true,
+                            "force reset counter",
+                            SegmentChangeRequestHistory.Counter.ZERO,
+                            ImmutableList.of()
+                        )
+                    )
+                )
+            ),
+            Futures.immediateFuture(
+                new ByteArrayInputStream(
+                    jsonMapper.writeValueAsBytes(
+                        new SegmentChangeRequestsSnapshot(
+                            false,
+                            null,
+                            SegmentChangeRequestHistory.Counter.ZERO,
+                            ImmutableList.of(
+                                new SegmentChangeRequestLoad(segment3),
+                                new SegmentChangeRequestLoad(segment4)
+                            )
+                        )
+                    )
+                )
+            )
+        )
+    );
+
+    DiscoveryDruidNode druidNode = new DiscoveryDruidNode(
+        new DruidNode("service", "host", 8080, null, new ServerConfig()),
+        DruidNodeDiscoveryProvider.NODE_TYPE_HISTORICAL,
+        ImmutableMap.of(
+            DataNodeService.DISCOVERY_SERVICE_KEY, new DataNodeService("tier", 1000, ServerType.HISTORICAL, 0)
+        )
+    );
+
+    HttpServerInventoryView httpServerInventoryView = new HttpServerInventoryView(
+        jsonMapper,
+        httpClient,
+        druidNodeDiscoveryProvider,
+        Predicates.alwaysTrue(),
+        new HttpServerInventoryViewConfig(null, null, null)
+    );
+
+    httpServerInventoryView.start();
```

Comment:
This should be stopped after test is finished.

Reply:
will add